### PR TITLE
Range-ify std.path.DirIteratorImpl

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -3047,14 +3047,23 @@ private struct DirIteratorImpl
...
-        if(stepIn(pathname))
+
+        static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
+            alias pathnames = pathname;
+        else
+        {
+            import std.array;
+            string pathnames = pathname.array;
+        }
+        if (stepIn(pathnames))
```
Comment (DmitryOlshansky) [inline std/file.d]: pathnames seems to indicate an array of path names which it isn't. Maybe pathnameStr or some such

**Comment by DmitryOlshansky** [comment]
LGTM otherwise

File: std/file.d (hunk 1 above)
Comment (andralex) [inline std/file.d]: yah the name needs changing

Comment (WalterBright) [inline std/file.d]: done

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by MartinNowak** [comment]
What about `dirEntries` and `DirIterator`?

**Pull request merged**